### DISCUSSION TITLE
Why are this code commented out?

### DISCUSSION THREAD

**Discussion by lbcsy** [project-jedi/jcl#141] [Q&A]
Why below code is commented out in JclSysUtils.pas? Is it safe to uncommented it?

```
(*
{$IFDEF SUPPORTS_GENERICS}
type
  ISafeGuard<T: class> = interface
    function ReleaseItem: T;
    function GetItem: T;
    procedure FreeItem;
    property Item: T read GetItem;
  end;

  TSafeGuard<T: class> = class(TObject, ISafeGuard<T>)
  private
    FItem: T;
    function ReleaseItem: T;
    function GetItem: T;
    procedure FreeItem;

  constructor Create(Instance: T);
  destructor Destroy; override;
  public
    class function New(Instance: T): ISafeGuard<T>; static;
  end;
{$ENDIF SUPPORTS_GENERICS}
*)
```

**Comment by obones** [comment]
I can see two possible reasons:

1. It crashes some older compiler
2. It brings no value whatsoever

**Reply by lbcsy** [reply]
>1. It crashes some older compiler
I believe this is not a problem for JCL as its installer has already been able to detect the version of Delphi installed.

So, the question will be "does generic style code bring any value"?

**Reply by lbcsy** [reply]
I uncommented the code and added back some missing code, then I can write some neat code like below:

```
procedure TForm2.Button1Click(Sender: TObject);
begin
  TSafeGuard<TForm3>.New(TForm3.Create(nil)).GetItem.ShowModal;
end;

```
no need try ... final.
